### PR TITLE
[SPARK-18092][ML] Fix column prediction type error

### PR DESCRIPTION
**Pull request by GnoDGnaF**
## What changes were proposed in this pull request?

Cast column prediction's type to Double in both `BinaryClassificationEvaluator` and `MulticlassClassificationEvaluator` 
## How was this patch tested?

manual tests

### CODE REVIEW

**Comment by d2Code** [comment]
cc @mengxr

File: mllib/src/main/scala/org/apache/spark/ml/evaluation/BinaryClassificationEvaluator.scala [hunk 1]
```diff
@@ -78,9 +78,11 @@ class BinaryClassificationEvaluator @Since("1.4.0") (@Since("1.4.0") override va
 
     // TODO: When dataset metadata has been implemented, check rawPredictionCol vector length = 2.
     val scoreAndLabels =
-      dataset.select(col($(rawPredictionCol)), col($(labelCol)).cast(DoubleType)).rdd.map {
+      dataset.select(col($(rawPredictionCol)), col($(labelCol)).cast(DoubleType))
+        .rdd
+        .map {
         case Row(rawPrediction: Vector, label: Double) => (rawPrediction(1), label)
-        case Row(rawPrediction: Double, label: Double) => (rawPrediction, label)
+        case Row(rawPrediction: Float, label: Double) => (rawPrediction.toDouble, label)
```
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/ml/evaluation/BinaryClassificationEvaluator.scala]: Why this change? in line 76 it's checked that it's a Double.

File: mllib/src/main/scala/org/apache/spark/ml/evaluation/MulticlassClassificationEvaluator.scala [hunk 2]
```diff
@@ -76,7 +76,9 @@ class MulticlassClassificationEvaluator @Since("1.5.0") (@Since("1.5.0") overrid
     SchemaUtils.checkNumericType(schema, $(labelCol))
 
     val predictionAndLabels =
-      dataset.select(col($(predictionCol)), col($(labelCol)).cast(DoubleType)).rdd.map {
+      dataset.select(col($(predictionCol)).cast(DoubleType), col($(labelCol)).cast(DoubleType))
```
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/ml/evaluation/MulticlassClassificationEvaluator.scala]: Ditto?

**Review by srowen — CHANGES_REQUESTED** [review]
